Paintroid/src/androidTest/java/org/catrobat/paintroid/test/espresso/tools/LineToolIntegrationTest.java

```diff
@@ -351,6 +369,8 @@ public void testConnectedLinesFeatureRedrawingLine() {
 
 	@Test
 	public void testClickingUndoOnceOnConnectedLines() {
+		CountingIdlingResource idlingResource = launchActivityRule.getActivity().getIdlingResource();
+		IdlingRegistry.getInstance().register(idlingResource);
```

Comment:
You already invoked .register() in the setup method, so I think there is no need to additionally .register() at the start of the test.